### DISCUSSION TITLE
Web Follower for Libre 3 - configuration

### DISCUSSION THREAD

**Discussion by sp6oje** [NightscoutFoundation/xDrip#2392] [Q&A]
Hi !. I am trying to config web follower and after step by step configuration I got no readings and in the status window
Last error: No preference value set for: password - cannot continue  ( labeled in red)

What wrong am I doing ? Where to search any errors ?

Could someone help me ?

Robert

**Comment by Navid200** [comment]
Have you entered the password under web follower settings?

**Reply by sp6oje** [reply]
Yes, I do. I tried to enter wrong password and then got Pprocess faulrue status A-Z {"status":2,"error":{"message":"motAuthenticated"}
Now after restoring right password I have no errors but did not get any data ... . LibreLinkUp show data but Xdrip Web Follower any.
In log now only information is : "Failed to get any response", but in LibreLinkUp I can see all my data
Where can I debug what is going wrong ?

**Reply by PhilTU** [reply]
Would removing Linkup (or Force Stop to test) allow data to flow to xDrip?

**Comment by Epsolone** [comment]
@Navid200 I have the same problem. I installed the latest version of xdrip+. When I tried set wrong credentials i had notAuthenticated error, but when i filled clear credentials i haven't errors but haven't data too. I tried with german account of LibreView, because now I'm in Germany. But my phone with English language. Is it can be the reason why app didn't try take data from German api?

Thanks.

**Reply by Navid200** [reply]
This capability was added knowing that it may be disabled.

When you post something like "I followed the instructions step by step", someone reading your post has no idea what instructions you have followed.  So, that statement doesn't really add any value.  You should either point to the instructions, or remove it from your statement.

**Reply by Epsolone** [reply]
But  I didn't post this request. I only found the same problem like i have. And asked about it you with some more details.

**Reply by Epsolone** [reply]
And i think that the author of this issue used this instruction https://xdrip.readthedocs.io/en/latest/install/webfollower/#configure-xdrip-web-follower

**Comment by calexwakefield** [comment]
I too have the exact same issue described by Robert here. I would love it if anyone found a solution.

**Comment by EarMaster** [comment]
Last night I suddenly got no readings from my Freestyle Libre 3 in xDrip any more. The Libre App itself works fine, uploads the data to LibreView, but there seems to be something off with the WebFollower connection. I checked the status page and there is an error:
```json
Pprocess failure status-AZ
{
  "status": 4,
  "data": {
    "step": {
      "type": "pp",
      "componentName": "AcceptDocument",
      "props": {
        "reaccept": true,
        "titleKey": "Common.privacyPolicy",
        "type": "pp"
      }
    }
    "user": {
      "accountType": "pat",
      "country": "DE",
      "uiLanguage": "de-DE"
    },
    "authTicket": {
      "token": "ey...",
      "expires": 1671535980,
      "duration": 3600000
    }
  }
}
```
I figured that there might be something related to updated privacy policies which I need to accept. So I logged into my account (the one I share the data with from my main account) and indeed I had to accept the privacy policy.

I did so, but unfortunately it didn't change anything in xDrip. I restarted the data collector, force quit and reopened xDrip and restarted my device without any effect. The error still persists and the last successful reading time was yesterday around 9pm.

```
Version: 344dev4-2022.11.16
Code: 2122111618
Downgradable to: 1603091400
```

**Reply by smbruck6** [reply]
> Last night I suddenly got no readings from my Freestyle Libre 3 in xDrip any more. The Libre App itself works fine, uploads the data to LibreView, but there seems to be something off with the WebFollower connection. I checked the status page and there is an error:
> 
> ```json
> Pprocess failure status-AZ
> {
>   "status": 4,
>   "data": {
>     "step": {
>       "type": "pp",
>       "componentName": "AcceptDocument",
>       "props": {
>         "reaccept": true,
>         "titleKey": "Common.privacyPolicy",
>         "type": "pp"
>       }
>     }
>     "user": {
>       "accountType": "pat",
>       "country": "DE",
>       "uiLanguage": "de-DE"
>     },
>     "authTicket": {
>       "token": "ey...",
>       "expires": 1671535980,
>       "duration": 3600000
>     }
>   }
> }
> ```
> 
> I figured that there might be something related to updated privacy policies which I need to accept. So I logged into my account (the one I share the data with from my main account) and indeed I had to accept the privacy policy.
> 
> I did so, but unfortunately it didn't change anything in xDrip. I restarted the data collector, force quit and reopened xDrip and restarted my device without any effect. The error still persists and the last successful reading time was yesterday around 9pm.
> 
> ```
> Version: 344dev4-2022.11.16
> Code: 2122111618
> Downgradable to: 1603091400
> ```
I am having the same issue

**Reply by PhilTU** [reply]
Has anyone checked if LibreLinkUp app is working with same credentials? The apps also ask for approval of T&Cs etc so perhaps there's a flag for that as well as for logging in at Libreview.com.

**Reply by EarMaster** [reply]
@PhilTU That did work! Thanks alot. It seems that the website and app don't share the same Terms & Conditions. Also the app doesn't require 2FA (while the website forces me to enter a code sent to my mail address). It is kind of worrying to see Abbott struggling to implement a proper security policy (although I suggest it helps us, because it saves us managing 2FA in xDrip+).

**Reply by Navid200** [reply]
@PhilTU @EarMaster Can I get an explanation of how you resolved the issue so that we can add it to documentation?

**Reply by klappstuhl** [reply]
Worked for me as well

@Navid200

I needed to logoff and logon in the libre link app. Otherwise i wont see the screen for the new licence agreements. As soon as i accepted the agreements xdrip worked fine

**Reply by Navid200** [reply]
@klappstuhl Thanks

**Reply by PhilTU** [reply]
> @PhilTU @EarMaster Can I get an explanation of how you resolved the issue so that we can add it to documentation?

There has been an app update for linkup v4.3.1(230040313) to 4.5.0 (230040515) which may be worth mentioning, these typically require acceptance of screens of terms & conditions. Not sure if @EarMaster noticed an update or just reopened app and accepted T&Cs.

**Reply by Navid200** [reply]
@PhilTU Thanks for resolving this.

I have updated this:  https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Reply by PhilTU** [reply]
@Navid200 thanks, may be worth adding something about  "I needed to logoff and logon in the libre link up app" - I just checked and without that you don't get offerred the terms & conditions pages.

**Reply by Navid200** [reply]
@PhilTU Do you mean if you have libre link uninstalled and you install it now, you will have to log out after just installing it and log in to get the terms and conditions?

**Reply by PhilTU** [reply]
> @PhilTU Do you mean if you have libre link uninstalled and you install it now, you will have to log out after just installing it and log in to get the terms and conditions?

I don't uninstall apps to update them personally, but having updated it was necessary to log out and in again to see the terms & conditions (at least on this occasion). I haven't tested the uninstall and reinstall scenario.

It was the LibreLink**Up** app that I was referring to, as the problem was with collecting the data. Librelink (or Libre 3 app) does the uploading to Libreview and that seemed to be ok.

**Reply by Navid200** [reply]
The instructions provided by the developer who wrote the web follow app into xDrip was that you could uninstall it after it was working.
But, looking at the instructions again, I can see that it does not ask you to uninstall it.  It was my mistake assuming that everyone would uninstall it reading the instructions.

So, I will add a note to highlight your point.  Thanks.

**Reply by davidcharlat** [reply]
the same problem happened to me 2 days ago ( Pprocess failure status-AZ{"status": 4,"data": {  etc... )
it has been solved by deconnecting me from librelinkup, reconnecting, accepting new terms of policy

**Reply by JPRecon** [reply]
Thanks David!  That fixed my issue too! LibreLinkUp app had an update.  Log out and back in, it forces you to accept new terms!

**Reply by davidcharlat** [reply]
you're wellcome. this probleme occured long time ago. Now I don't use librelink anymore, I'm using Juggluco instead. Juggluco sends my blood sugar values to librelinkup (for those who want them). i haven't known any problem since the day i installed juggluco
